docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,192 @@
+.. _chat_dataset_usage_label:
+
+=============
+Chat Datasets
+=============
+
+Chat datasets involve multi-turn conversations (multiple back-and-forths) between user and assistant.
+This is more structured than freeform text association that models are typically pre-trained with,
+where they learn to simply predict the next token instead of responding accurately to the user.
+
+The primary entry point for fine-tuning with chat datasets in torchtune is the :func:`~torchtune.datasets.chat_dataset`
+builder. This lets you specify a local or Hugging Face dataset that follows the chat data format
+directly from the config and train your LLM on it.
+
+Chat dataset format
+-------------------
+
+Chat datasets typically have a single column named "conversations" or "messages" that contains a list of messages on a single topic
+per sample. The list of messages could include a system prompt, multiple turns between user and assistant, and tool calls/returns.
+
+.. code-block:: text
+
+    |  conversations                         |
+    |----------------------------------------|
+    | [{"role": "user", "content": Q1},      |
+    |  {"role": "assistant", "content": A1}] |
+
+The conversation structure in the dataset will be reformatted to follow torchtune's :class:`~torchtune.data.Message` structure.
+The above will be converted to:
+
+.. code-block:: python
+
+    messages = [
+        Message(role="user", content="Q1"),
+        Message(role="assistant", content="A1"),
+    ]
+
+The list of messages is tokenized by the model tokenizer with the appropriate model-specific special tokens added
+(such as beginning-of-sequence, end-of-sequence, and others).
+
+.. code-block:: python
+
+    from torchtune.models.phi3 import phi3_mini_tokenizer
+
+    p_tokenizer = phi3_mini_tokenizer("/tmp/Phi-3-mini-4k-instruct/tokenizer.model")
+    tokens, mask = p_tokenizer.tokenize_messages(messages)
+    print(tokens)
+    # [1, 32010, 29871, 13, 29984, 29896, 32007, 29871, 13, 32001, 29871, 13, 29909, 29896, 32007, 29871, 13]
+    print(p_tokenizer.decode(tokens))
+    # '\nQ1 \n \nA1 \n'
+
+As an example, you can see the schema of the `SlimOrca dataset <https://huggingface.co/datasets/Open-Orca/SlimOrca-Dedup>`_.
+
+Specifying conversation style
+-----------------------------
+
+The structure of the conversation in the raw dataset can vary widely with different role names, different fields indicating
+where the message content name, and other ways. There are a few standardized formats that are common across many datasets.
+We have built-in converters to convert these standardized formats into a list of torchtune :class:`~torchtune.data.Message`
+that follows this format:
+
+.. code-block:: python
+
+    [
+        {
+            "role": "system" | "user" | "assistant" | "ipython",
+            "content": <message>,
+        },
+        ...
+    ]
```

Comment:
I'm a bit on the fence about this way of representing the different chat formats. While it's very general and the most concise I think it takes a second of staring at it to understand. Anyways I don't have a real ask to change this rn, but we should think about whether it makes sense to give explicit examples here as well.